pkg/discovery/discovery.go

```diff
@@ -304,6 +305,17 @@ func (d *Discoverer) createPV(file, class string, reclaimPolicy v1.PersistentVol
 	klog.Infof("Found new volume at host path %q with capacity %d, creating Local PV %q, required volumeMode %q",
 		outsidePath, capacityByte, pvName, volMode)
 
+	pvLabels := make(map[string]string)
+	for k, v := range d.Labels {
+		pvLabels[k] = v
+	}
+	// truncate the class name if it exceeds the max allowed length
+	if len(class) > validation.LabelValueMaxLength {
+		pvLabels[common.LocalStorageClassLabel] = class[:validation.LabelValueMaxLength]
+	} else {
+		pvLabels[common.LocalStorageClassLabel] = class
+	}
```

Comment:
The PV will be rejected by the API server with a long label value (> 63 chars). Because the storage class name is for information only, it's safe to truncate it.

Reply:
Thats a good point. I think I should have done this for user configured labels that I added in my previous PR. Can we fix that here too or do you want me to open a follow up PR?

Reply:
I don't think it is necessary for user-configured labels. Users should fix their labels if the local-volume-provisioner cannot create PVs, otherwise, they may fail to list PVs with their PV labels.